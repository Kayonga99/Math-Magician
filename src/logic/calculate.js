import operate from './operate';

function isNumber(item) {
  return !!item.match(/[0-9]+/);
}

/**
 * Given a button name and a calculator data object, return an updated
 * calculator data object.
 */
export default function calculate(obj, buttonName) {
  if (buttonName === 'AC') {
    return {
      total: null,
      next: null,
      operation: null,
    };
  }

  if (isNumber(buttonName)) {
    if (buttonName === '0' && obj.next === '0') {
      return {};
    }
    // If there is an operation, update next
    if (obj.operation) {
      if (obj.next && obj.next !== '0') {
        return { ...obj, next: obj.next + buttonName };
      }
      return { ...obj, next: buttonName };
    }
    // If there is no operation, update next and clear the value
    if (obj.next && obj.next !== '0') {
      return {
        next: obj.next + buttonName,
        total: null,
      };
    }
    return {
      next: buttonName,
      total: null,
    };
  }
