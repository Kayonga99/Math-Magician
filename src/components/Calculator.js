import React from 'react';
import './calculatorStyles.css';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    this.setState((state) => calculate(state, e.target.id));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="calculator">
        <h2 className="output c-item">
          <span>{total}</span>
          <span>{operation}</span>
          <span>{next}</span>
        </h2>
        <button type="button" onClick={this.handleClick} id="AC" className="c-item">AC</button>
        <button type="button" onClick={this.handleClick} id="+/-" className="c-item">+/-</button>
        <button type="button" onClick={this.handleClick} id="%" className="c-item">%</button>
        <button type="button" onClick={this.handleClick} id="รท" className="c-item bg-orange">รท</button>
        <button type="button" onClick={this.handleClick} id="7" className="c-item">7</button>
        <button type="button" onClick={this.handleClick} id="8" className="c-item">8</button>
        <button type="button" onClick={this.handleClick} id="9" className="c-item">9</button>
        <button type="button" onClick={this.handleClick} id="x" className="c-item bg-orange">x</button>
        <button type="button" onClick={this.handleClick} id="4" className="c-item">4</button>
        <button type="button" onClick={this.handleClick} id="5" className="c-item">5</button>
        <button type="button" onClick={this.handleClick} id="6" className="c-item">6</button>
        <button type="button" onClick={this.handleClick} id="-" className="c-item bg-orange">-</button>
        <button type="button" onClick={this.handleClick} id="1" className="c-item">1</button>
        <button type="button" onClick={this.handleClick} id="2" className="c-item">2</button>
        <button type="button" onClick={this.handleClick} id="3" className="c-item">3</button>
        <button type="button" onClick={this.handleClick} id="+" className="c-item bg-orange">+</button>
        <button type="button" onClick={this.handleClick} id="0" className="c-item zero">0</button>
        <button type="button" onClick={this.handleClick} id="." className="c-item">.</button>
        <button type="button" onClick={this.handleClick} id="=" className="c-item bg-orange">=</button>
      </div>
    );
  }
}

export default Calculator;
