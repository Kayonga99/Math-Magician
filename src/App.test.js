

import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import Calculator from './components/Calculator';
import Home from './components/home';
import NavBar from './components/navbar';
import Quotes from './components/quote';
import calculate from './logic/calculate';
import operate from './logic/operate';