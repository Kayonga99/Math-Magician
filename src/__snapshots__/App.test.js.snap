// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component test Calculator component should perform the calculations correctly on initial run 1`] = `
<div
  className="calculator"
>
  <h2
    className="output c-item"
  >
    <span>
      0
    </span>
    <span />
    <span />
  </h2>
  <button
    className="c-item"
    id="AC"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="c-item"
    id="+/-"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="c-item"
    id="%"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="c-item bg-orange"
    id="รท"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="c-item"
    id="7"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="c-item"
    id="8"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="c-item"
    id="9"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="c-item bg-orange"
    id="x"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="c-item"
    id="4"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="c-item"
    id="5"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="c-item"
    id="6"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="c-item bg-orange"
    id="-"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="c-item"
    id="1"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="c-item"
    id="2"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="c-item"
    id="3"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="c-item bg-orange"
    id="+"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="c-item zero"
    id="0"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="c-item"
    id="."
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="c-item bg-orange"
    id="="
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`component test Home component should perform correctly on initial run 1`] = `
<div
  className="homeParagraph"
>
  <h2>
    Welcome to our page!
  </h2>
  <br />
  <br />
  <p
    className="paragraph"
  >
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
  </p>
  <br />
  <br />
  <p
    className="paragraph"
  >
    "Lorem ipsum dolor sit amet,Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
  </p>
</div>
`;

exports[`component test NavBar component should perform correctly on initial run 1`] = `
<header
  className="nav"
>
  <h1>
    Math Magicians
  </h1>
  <nav
    className="nav-links"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/calculator"
      onClick={[Function]}
    >
       Calculator
    </a>
    <a
      href="/quotes"
      onClick={[Function]}
    >
       Quote
    </a>
  </nav>
</header>
`;

exports[`component test Quote component should perform correctly on initial run 1`] = `
<main
  className="quotesParagraph"
>
  <p
    className="paragraphQuote"
  >
    Mathematics is not about numbers, equations, computations or algorithms: It is about understanding. - William Paul, Thurston
  </p>
</main>
`;
